src/tuple_utils.c

```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright (c) 2016-2018  Timescale, Inc. All Rights Reserved.
+ *
+ * This file is licensed under the Apache License,
+ * see LICENSE-APACHE at the top level directory.
+ */
+#include <postgres.h>
+#include <access/htup_details.h>
+
+#include "tuple_utils.h"
+
+/* Create a fresh struct pointer that will contain copied contents of the tuple */
+void *
+create_struct_from_tuple(HeapTuple tuple, MemoryContext mctx, size_t alloc_size, size_t copy_size)
+{
+	void	   *struct_ptr = MemoryContextAllocZero(mctx, alloc_size);
```

Comment:
It'll probably give better valgrind errors if this is just `MemoryCotextAlloc` not `MemoryContextAllocZero `.